tools/block-generator/generator/generate.go

```diff
@@ -763,110 +876,242 @@ func (g *generator) generateAssetTxnInternalHint(txType TxTypeID, round uint64,
 		fmt.Printf("\n\nthe sender account does not have enough algos for the transfer. idx %d, asset transaction type %v, num %d\n\n", senderIndex, actual, g.reportData[actual].GenerationCount)
 		os.Exit(1)
 	}
+
+	if assetID == 0 {
```

Comment:
when do you expect this to happen?

Reply:
This only happens if the assetID didn't get assigned correctly. That is, if I introduced a bug.

Reply:
I'll add "this should never happen"

Reply:
ae6d9108f08303028e1ec043b3741a9b49af4240